app/controllers/events_controller.rb

```diff
@@ -0,0 +1,57 @@
+class EventsController < ApplicationController
+  def index
+    @events = Event.order(:date)
+  end
+
+  def show
+    @event = Event.find(params[:id])
+  end
+
+  def new
+    @event = Event.new
+  end
+
+  def create
+    @event = Event.new(event_params)
+    if @event.save
+      flash[:notice] = "Event Created Successfully"
+      redirect_to(events_path)
+    else
+      render('new')
+    end
+  end
+
+  def edit
+    @event = Event.find(params[:id])
+  end
+
+  def postpone
+    @event = Event.find(params[:id])
```

Comment:
Is there a postpone check to see if the user postpones and sets a `null` date? You do this for `edit` because `update` sends you to `edit` on fail. Perhaps add a new controller for fail on `postpone`?